code/game/machinery/computer/ai_core.dm

```diff
@@ -254,6 +254,42 @@ That prevents a few funky behaviors.
 								to_chat(U, "<span class='danger'>ERROR:</span> Reconstruction in progress.")
 							else if (!T.occupant)
 								to_chat(U, "<span class='danger'>ERROR:</span> Unable to locate artificial intelligence.")
+			if("AIUPLOAD")//AI Upload terminal.
+				var/obj/machinery/computer/aiupload/T = target
+				switch(interaction)
+					if("AICARD")
+						var/obj/item/device/aicard/C = src
+						if(!T.contents.len)
+							if (!C.contents.len)
+								to_chat(U, "No AI to copy over!")//Well duh
+
+							else
+								for(var/mob/living/silicon/ai/A in C)
+									C.icon_state = "aicard"
+									C.name = "inteliCard"
+									C.overlays.len = 0
+									A.forceMove(T)
+									T.occupant = A
+									A.control_disabled = 1
+									A.cancel_camera()
+									to_chat(A, "You have been uploaded to a stationary terminal. Sadly, there is no remote access from here.")
+									to_chat(U, "<span class='notice'><b>Transfer successful</b>:</span> [A.name] ([rand(1000,9999)].exe) installed and executed successfully. Local copy has been removed.")
+						else
+							if(!C.contents.len && T.occupant)
+								C.name = "inteliCard - [T.occupant.name]"
+								if (T.occupant.stat == 2)
+									C.icon_state = "aicard-404"
+								else
+									C.icon_state = "aicard-full"
+								to_chat(T.occupant, "You have been downloaded to a mobile storage device. Still no remote access.")
```

Comment:
It may have remote access. Check `T.occupant.control_disabled`

Reply:
ive added a check that tells the AI if they have remote access upon being transferred back into the card